packages/klighd-core/src/views-rendering.tsx

```diff
@@ -477,6 +489,37 @@ export function renderKText(rendering: KText, parent: SKGraphElement | SKLabel,
     const shadowStyles = getSvgShadowStyles(styles, context)
     const textStyles = getSvgTextStyles(styles)
 
+    // Replace text with rectangle, if the text is too small.
+    const region = context.depthMap?.getProvidingRegion(parent as KNode, context.viewport, context.renderingOptions)
+
+    const simplifySmallTextOption = context.renderingOptions.getValueForId(SimplifySmallText.ID)
+    const simplifySmallText = simplifySmallTextOption ?? false // Only enable, if option is found.
+    if (simplifySmallText && (!region || region.detail === DetailLevel.FullDetails) && !rendering.isNodeTitle) {
+        const simplificationThresholdOption = context.renderingOptions.getValueForId(TextSimplificationThreshold.ID)
+        const defaultThreshold = 3
```

Comment:
defaults should be stored in a more central location and not as some magic number in the rendering code. For this example maybe as TextSimplificationThreshold.DEFAULT